firmware/controllers/serial/serial_sensor.cpp

```diff
@@ -0,0 +1,175 @@
+/**
+ * @file	serial_sensor.cpp
+ *
+ *
+ * @date Mar 19, 2020
+ * @author Konstantin Smola, (c) 2020
+ */
+
+#include "global.h"
+#if EFI_AUX_SERIAL
+#include "serial.h"
+#include "serial_sensor.h"
+#include "engine.h"
+
+#define NUM_INNOVATE_O2_SENSORS 1
+#define AFR_MULTIPLIER 147
+
+EXTERN_ENGINE;
+
+volatile float InnovateLC2AFR = AFR_ERROR;
+
+typedef enum
+{
+	NO_ERROR = 0,
+	HEATER_SHORTED = 1,
+	HEATER_OPEN = 2,
+	PUMPCELL_SHORTED = 3,
+	PUMPCELL_OPEN = 4,
+	REFCELL_SHORTED = 5,
+	REFCELL_OPEN = 6,
+	SYSTEM_ERROR = 7,
+	SENSOR_TIMING_ERR = 8,
+	SUPP_V_LOW = 9
+} sensor_error_code_t;
+
+typedef struct
+{
+	int function_code;
+	float AFR;
+	float AFR_multiplier;
+	float lambda;
+	float warmup;
+	sensor_error_code_t error_code;
+} sensor_data_t;
+
+static sensor_data_t innovate_o2_sensor[NUM_INNOVATE_O2_SENSORS];
+
+static size_t tmsglen;
+
+void IdentifyInnovateSerialMsg() {		//this identifies an innovate LC1/LC2 o2 sensor by it's first word (header)
+	if (CONFIG(enableInnovateLC2)) {
+		if (((ser_buffer[0]) & 162) != 162) {		//not serial header word
```

Comment:
I see "162" multiple times and I thinking "Oh my god, that's a magic constant! It MUST mean something?!"

https://en.wikipedia.org/wiki/Magic_number_(programming)

Reply:
`constexpr uint8_t innovateHeaderValue = 162`